python/grass/pygrass/vector/abstract.py

```diff
@@ -413,14 +413,33 @@ def open(
             str_err = "Not able to open the map, C function return %d."
             raise OpenError(str_err % openvect)
 
+        # Load attribute table for selected layer.
         if len(self.dblinks) == 0:
             self.layer = layer
             self.table = None
             self.n_lines = 0
         else:
-            self.layer = self.dblinks.by_layer(layer).layer
-            self.table = self.dblinks.by_layer(layer).table()
+            layer_db_link = self.dblinks.by_layer(layer)
+            if not layer_db_link:
+                raise LookupError(
+                    "There appears to be no database link for layer %d of <%s>."
+                    % (layer, self.name)
+                )
```

Comment:
Previously, this would error because it would try to access `None.layer`, so I kept the general behavior of producing an exception. However, if there are zero links (case above) then it is OK to just not load the table, so this may still be inconsistent behavior.